catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/JsonPersistantStore.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Reader;
+import java.nio.file.Paths;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonPersistantStore implements ObjectPersistentStore {
+
+  private Gson gson = new Gson();
+
+  private final String mapName;
+
+  public JsonPersistantStore() {
+    mapName = getClass().getSimpleName();
+  }
+
+  public JsonPersistantStore(String mapName) {
+    this.mapName = mapName;
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JsonPersistantStore.class);
+
+  private static final String PERSISTED_FILE_SUFFIX = ".json";
+
+  private String getShaFor(String key) {
+    return DigestUtils.sha1Hex(key);
+  }
+
+  private String getPath() {
```

Comment:
❗️Instead of converting the `Path` to a `String`, we should just return the `Path` object and have the callers use that instead. That way no one should have to deal with file separators and can use existing `Path` methods.